controllers/admin/AdminPsThemeCustoAdvanced.php

```diff
@@ -80,18 +83,100 @@ public function initContent()
     */
     public function ajaxProcessDownloadChildTheme()
     {
-        global $kernel;
+        $bPrepareChildtheme = self::prepareChildTheme(_THEME_NAME_, _PS_THEME_DIR_);
 
-        if (!$this->module->hasEditRight()) {
-            return $this->l("You do not have permission to edit this.");
+        if (!$bPrepareChildtheme) {
+            die(false);
+        }
+
+        $bCreateChildTheme = self::createChildTheme(_THEME_NAME_);
+
+        if (!$bCreateChildTheme) {
+            die(false);
+        }
+
+        die(self::getChildTheme(_THEME_NAME_, _PS_ROOT_DIR_));
+    }
+
+    /**
+     * Prepare the child theme
+     *
+     * @param string $sParentThemeName
+     * @param string $sParentThemeDir
+     * @return bool
+    */
+    private function prepareChildTheme($sParentThemeName, $sParentThemeDir)
+    {
+        Tools::ZipExtract($this->childtheme_skeleton, $this->sandbox_path);
+
+        $aStringToReplace = array(
+            '{childtheme_parent}' => $sParentThemeName,
+            '{childtheme_name}' => 'child_'.$sParentThemeName,
+            '{childtheme_description}' => 'Child theme of '.$sParentThemeName.'\'s theme'
+        );
+
+        $sChildThemeConfigPath = $this->sandbox_path.'/'.$this->skeleton_name.'/config/theme.yml';
+
+        $sConfigFile = @file_get_contents($sChildThemeConfigPath);
+
+        foreach ($aStringToReplace as $sSearchElement => $sReplace) {
+            $sConfigFile = str_replace($sSearchElement, $sReplace, $sConfigFile);
+        }
+
+        $bPutContents = @file_put_contents($sChildThemeConfigPath, $sConfigFile);
+
+        if (!$bPutContents) {
+            return false;
         }
 
-        $exporter = $kernel->getContainer()->get('prestashop.core.addon.theme.exporter');
-        $path = $exporter->export($this->context->shop->theme);
-        $aPath = array_reverse(explode('/', $path));
-        $sThemeZipPath = $this->module->ps_uri.'/themes/'.$aPath[0];
+        $bCopyFile = @copy($sParentThemeDir.'/preview.png', $this->sandbox_path.'/'.$this->skeleton_name.'/preview.png');
+
+        if (!$bCopyFile) {
+            return false;
+        }
 
-        die($sThemeZipPath);
+        return true;
+    }
+
+    /**
+     * Create the child theme
+     *
+     * @param string $sParentThemeName
+     * @param string $dest
+     * @return bool
+    */
+    private function createChildTheme($sParentThemeName)
+    {
+        $sChildThemeFolderName = 'child_'.$sParentThemeName;
+
+        $oZip = new ZipArchive;
+        $oZip->open($this->sandbox_path.'/'.$sChildThemeFolderName.'.zip', ZipArchive::CREATE);
```

Comment:
@Adrienaddons is it safe to use `/` directly? shouldn't we have `DIRECTORY_SEPARATOR`? cc @Quetzacoalt91